### PR TITLE
bridge: Fix disk sample ignoring of device-mapper; adjust TestWireGuard for Ubuntu 23.04

### PR DESCRIPTION
**Pull request by martinpitt**
See https://github.com/cockpit-project/bots/pull/5432 for details, especially around the device mapper change [debug info](https://github.com/cockpit-project/bots/pull/5432#issuecomment-1849471326).

### CODE REVIEW

File: src/cockpit/samples.py [hunk 1]
```diff
@@ -228,12 +228,12 @@ def sample(self, samples: Samples) -> None:
...
-                # ignore device-mapper and md
-                if dev_major in ['9', '253']:
+                # ignore mdraid
+                if dev_major == '9':
                     continue
 
-                # Skip partitions
-                if dev_name[:2] in ['sd', 'hd', 'vd'] and dev_name[-1].isdigit():
+                # Skip partitions and device mapper
+                if dev_name[:2] in ['sd', 'hd', 'vd', 'md'] and dev_name[-1].isdigit():
```
Comment (mvollmer) [inline src/cockpit/samples.py:236]: "dm"?

**Review by mvollmer — CHANGES_REQUESTED** [review]

File: src/cockpit/samples.py (hunk 1 above)
Comment (mvollmer) [reply src/cockpit/samples.py:236]: I think the devicemapper check should be `dev_name[:3] == "dm-"`, just to be clearer.

**Review by mvollmer — CHANGES_REQUESTED** [review]

**Comment by martinpitt** [comment]
For the remaining TestWireGuard.testVPN failure, the problem is a wrongly written netmask. On F39 the config file and actual state have `allowed-ips=10.0.0.1/32;2001::1`, while /etc/netplan/90-NM-e67d4dcf-d4f1-4566-8eff-fb7fcc72a795.yaml adds a default netmask of `"2001::1/32"` which is bogus for IPv6. It can be worked around with

```diff
-        b2.set_input_text("#network-wireguard-settings-allowedips-peer-0", f"{m1_ip4}/32,{m1_ip6}")
+        b2.set_input_text("#network-wireguard-settings-allowedips-peer-0", f"{m1_ip4}/32,{m1_ip6}/128")
```

but let's not do that. I'll report a bug instead.

**Review by mvollmer — APPROVED** [review]
Dänks!

**Pull request merged**